beacon-chain/rpc/eth/validator/validator.go

```diff
@@ -109,9 +109,15 @@ func (vs *Server) GetAttesterDuties(ctx context.Context, req *ethpbv1.AttesterDu
 		return nil, status.Errorf(codes.Internal, "Could not get dependent root: %v", err)
 	}
 
+	isOptimistic, err := rpchelpers.IsOptimistic(ctx, s, vs.HeadFetcher)
```

Comment:
You want to move this before `s, err = advanceState(ctx, s, req.Epoch, currentEpoch)` or else state `s` gets mutated